polars/polars-lazy/polars-plan/src/logical_plan/builder.rs

```diff
@@ -253,7 +253,15 @@ impl LogicalPlanBuilder {
         try_parse_dates: bool,
     ) -> PolarsResult<Self> {
         let path = path.into();
-        let mut file = std::fs::File::open(&path)?;
+        let mut file = std::fs::File::open(&path).map_err(|e| {
```

Comment:
Can you make this as an `open_file() -> PolarsResult<File>` function in `polars-utils` and use that function everywhere we open a file and return an error if it not exists?